src/Marten/Internal/Sessions/DocumentSessionBase.cs

```diff
@@ -319,26 +319,37 @@ public void EjectPatchedTypes(IUnitOfWork changes)
             foreach (var type in patchedTypes) EjectAllOfType(type);
         }
 
-        internal interface IHandler
+        internal interface IObjectHandler
```

Comment:
The interface name change from `IHandler` to `IObjectHandler` is okay. But I felt that it could have been retained as `IHandler` itself (nitpicking here)

Reply:
I do agree, and I did actually consider this while making the change. My intent was to make the interface more specific to the type handled. However, I'm not opposed to changing this back to `IHandler` if anyone feels strongly about this.